## Etapa 1: Compreender

### Libraries 

In [2]:
import pandas as pd
import numpy as np

### Read

In [5]:
df = pd.read_csv('tempo.csv', delimiter=";")
df

Aparencia  Temperatura  Umidade       Vento Jogar
0        sol           85     85.0       FALSO   nao
1        sol           80     90.0  VERDADEIRO   nao
2    nublado           83     86.0       FALSO   sim
3      chuva           70      NaN       FALSO   sim
4      chuva           68     80.0       FALSO   sim
5      chuva           65     70.0  VERDADEIRO   nao
6      menos           64     65.0  VERDADEIRO   sim
7        sol         1220     95.0       FALSO   nao
8        sol           69     70.0       FALSO   sim
9      chuva           75    200.0         NaN   sim
10       sol           75     70.0  VERDADEIRO   sim
11   nublado           72     90.0  VERDADEIRO   sim
12   nublado           81     75.0       FALSO   sim
13     chuva           71     91.0  VERDADEIRO   nao

Considerações: antes de prosseguir, já identifiquei:
* na coluna aparência tem um dado categorico "menos", onde deveria ser sol, nublado, chuva
* na coluna temperatura tem uma temperatura 1220 indicando outlier
* Posso criar uma coluna de temperatura em graus Célsius () → uma função também?
* na coluna umidade tem NaN; tem 200, isso é um outlier?
* em vento tem NaN
* Jogar significa se dá para jogar bola no dia em questão para fazer possíveis previsões. (quais padrões?)

### Shape 

In [9]:
df.shape

(14, 5)

Considerações: (14 linhas [contanto o cabeçalho], 5 colunas [não contanto o index do pandas])

### Head e Tail

In [12]:
df.head()

Aparencia  Temperatura  Umidade       Vento Jogar
0       sol           85     85.0       FALSO   nao
1       sol           80     90.0  VERDADEIRO   nao
2   nublado           83     86.0       FALSO   sim
3     chuva           70      NaN       FALSO   sim
4     chuva           68     80.0       FALSO   sim

In [13]:
df.tail()

Aparencia  Temperatura  Umidade       Vento Jogar
9      chuva           75    200.0         NaN   sim
10       sol           75     70.0  VERDADEIRO   sim
11   nublado           72     90.0  VERDADEIRO   sim
12   nublado           81     75.0       FALSO   sim
13     chuva           71     91.0  VERDADEIRO   nao

### Columns

In [15]:
df.columns

Index(['Aparencia', 'Temperatura', 'Umidade', 'Vento', 'Jogar'], dtype='object')

### Info

In [19]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14 entries, 0 to 13
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Aparencia    14 non-null     object 
 1   Temperatura  14 non-null     int64  
 2   Umidade      13 non-null     float64
 3   Vento        13 non-null     object 
 4   Jogar        14 non-null     object 
dtypes: float64(1), int64(1), object(3)
memory usage: 692.0+ bytes


### Describe (números)

In [21]:
df.describe()

Temperatura     Umidade
count    14.000000   13.000000
mean    155.571429   89.769231
std     306.433759   34.540203
min      64.000000   65.000000
25%      69.250000   70.000000
50%      73.500000   85.000000
75%      80.750000   90.000000
max    1220.000000  200.000000

### Unique (categorias) 

In [27]:
pd.unique(df)

C:\Users\guifp\AppData\Local\Temp\ipykernel_14636\656283662.py:1: FutureWarning: unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
  pd.unique(df)


ValueError: could not broadcast input array from shape (14,5) into shape (14,)

## Etapa 2: Limpar

### Colunas inúteis e/ou redundantes

### Linhas duplicadas e/ou redundantes 

### Outliers ou valores impossíveis 

### Verificar Valores Ausentes: isnull 

### Tratar Valores Ausentes: 

Remover Linhas com Valores Ausentes: dropna 

Substituir Valores Ausentes: fillna(0) 

### Conversão de Tipos de Dados

### Remover ruídos 

## Etapa 3: Relacionar

### Análise univariada:
Foca em uma única variável, ajudando a entender sua distribuição e características básicas.  


#### Distribuição de Frequências

#### Medidas de Tendência Central

#### Medidas de Dispersão

#### Q-Q Plot

### Análise bivariada:
Explora a relação entre duas variáveis, buscando padrões e correlações.  


#### Tabela de Contingência

#### Correlação Pearson

#### Correlação Spearman

### Análise multivariada: 
Investiga múltiplas variáveis simultaneamente para identificar interações complexas. 

#### Análise de Regressão (linear ou logística)

#### Análise de Componentes Principais (PCA)

#### Clusterização (K-means)